odl/deform/optimal_information_transport.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Solver for the shape-based reconstruction using
+optimal information transportation.
+
+The model is:
+
+min sigma * (1 - sqrt{DetJacInvPhi})^2 + (T(phi.I) - g)^2,
+where phi.I := DetJacInvPhi * I(InvPhi) is a mass-preserving deformation.
+
+Note that:
+If T is an identity operator, the above model reduces for image matching.
+If T is a forward projection operator, the above model is
+for image reconstruction.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+import numpy as np
+from odl.discr import Gradient
+from odl.trafos import FourierTransform
+standard_library.install_aliases()
+
+__all__ = ('optimal_information_transport_solver',)
+
+
+def optimal_information_transport_solver(gradS, I, niter, eps,
+                                         sigma, callback=None):
+
```

Comment:
We should now add documentation here, similar to the doc in [chambolle pock](https://github.com/odlgroup/odl/blob/3543720d91841c602bfebf94bdbe26986a6f3718/odl/solvers/advanced/chambolle_pock.py#L43-L123)